comps/imagegen/imagegen.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import base64
+import os
+import sys
+import time
+from io import BytesIO
+
+import numpy as np
+import tritonclient.http as httpclient
+from PIL import Image
+from tritonclient.utils import *
+
+from comps import Base64ByteStrDoc, ServiceType, TextDoc, opea_microservices, opea_telemetry, register_microservice
+
+
+@opea_telemetry
+def generate_image(*, text, triton_endpoint):
+    start = time.time()
+
+    with httpclient.InferenceServerClient(triton_endpoint, network_timeout=1000 * 300) as client:
```

Comment:
What's the reasoning for the `300s` timeout? Isn't that a bit too long?

Reply:
The initial ImageGen request can take a while - having the ImageGen non-model comp "warm up" the model service would let us reduce this, is that a pattern seen in any other comp I can lift?